File: arrays/validate-subsequence/validate-subsequence.py
array = [5, 1, 22, 25, 6, -1, 8, 10]
sequence = [1, 6, -1, 10]


# def validate_subsequence(array, sequence):
#     array_idx = 0
#     seq_idx = 0
#     while array_idx < len(array) and seq_idx < len(sequence):
#         if array[array_idx] == sequence[seq_idx]:
#             seq_idx += 1
#         array_idx += 1
#     return seq_idx == len(sequence)


def validate_subsequence(array, sequence):
    seq_idx = 0
    for arr in array:
        if seq_idx == len(sequence):
            break
        if sequence[seq_idx] == arr:
            seq_idx += 1
    return seq_idx == len(sequence)


print(validate_subsequence(array, sequence))

File: arrays/validate-subsequence/validate-subsequence.go
package main

import "fmt"

func main() {
	array := []int{5, 1, 22, 25, 6, -1, 8, 10}
	sequence := []int{1, 6, -1, 10}
	fmt.Println(validateSubsequence(array, sequence))

}

// func validateSubsequence(array, sequence []int) bool {
// 	var arrayIdx int
// 	var seqIdx int

// 	for arrayIdx < len(array) && seqIdx < len(sequence) {
// 		if array[arrayIdx] == sequence[seqIdx] {
// 			seqIdx += 1
// 		}
// 		arrayIdx += 1
// 	}

// 	return seqIdx == len(sequence)
// }

func validateSubsequence(array, sequence []int) bool {
	var seqIdx int

	for _, arr := range array {
		if seqIdx == len(sequence) {
			break
		}
		if arr == sequence[seqIdx] {
			seqIdx += 1
		}
	}

	return seqIdx == len(sequence)
}
